df_engine_extensions/src/dist_sql_query/physical_plan.rs

```diff
@@ -147,31 +151,45 @@ pub(crate) struct ResolvedPartitionedScan {
     pub remote_exec_ctx: Arc<RemoteExecContext>,
     pub pushdown_continue: bool,
     pub metrics_collector: MetricsCollector,
+    pub remote_metrics_collector: HashMap<String, RemoteMetricsCollector>,
```

Comment:
As I think, you can define somethings like `detailed_remote_metrics` whose type can be `Arc<Vec<Mutex<String>>>`, and place it into `SubTablePlanContext`.
Then pass it from `SubTablePlanContext` to `PartitionedStream`, and set related metrics to it returned from stream of `RemotePhysicalPlanExecutor`.
Finally, display it through `ResolvedPartitionedScan`.